### ISSUE TITLE
No volume repair mechanism for image recovery after sled-agent restart

### ISSUE THREAD

**Issue by askfongjojo** [oxidecomputer/crucible#862]
I have 2 images which have a downstair copy on the sleds on which the sled-agents were restarted. Disks created out of these images reported successful completion but instances attaching to them failed to boot. I inspected the propolis log for one of such instances and it seems to be not making any progress - I only saw log lines like these ones:
```
{"msg":"[1] Received Imok","v":0,"name":"propolis-server","level":30,"time":"2023-07-30T00:51:42.909133304Z","hostname":"oxz_propolis-server_c08fc106-9de9-4679-959b-cb66fe71644d","pid":4713,"component":"crucible-6a9b2a6f-0501-4be4-9679-cceb4c893b09"}
{"msg":"[2] Received Imok","v":0,"name":"propolis-server","level":30,"time":"2023-07-30T00:51:44.470049589Z","hostname":"oxz_propolis-server_c08fc106-9de9-4679-959b-cb66fe71644d","pid":4713,"component":"crucible-6a9b2a6f-0501-4be4-9679-cceb4c893b09"}
{"msg":"[1] Received Imok","v":0,"name":"propolis-server","level":30,"time":"2023-07-30T00:51:47.910901299Z","hostname":"oxz_propolis-server_c08fc106-9de9-4679-959b-cb66fe71644d","pid":4713,"component":"crucible-6a9b2a6f-0501-4be4-9679-cceb4c893b09"}
{"msg":"[2] Received Imok","v":0,"name":"propolis-server","level":30,"time":"2023-07-30T00:51:49.471338429Z","hostname":"oxz_propolis-server_c08fc106-9de9-4679-959b-cb66fe71644d","pid":4713,"component":"crucible-6a9b2a6f-0501-4be4-9679-cceb4c893b09"}
{"msg":"[0] connecting to [fd00:1122:3344:107::b]:19001","v":0,"name":"propolis-server","level":30,"time":"2023-07-30T00:51:49.762448696Z","hostname":"oxz_propolis-server_c08fc106-9de9-4679-959b-cb66fe71644d","pid":4713,"looper":"0","component":"crucible-6a9b2a6f-0501-4be4-9679-cceb4c893b09"}
{"msg":"[1] Received Imok","v":0,"name":"propolis-server","level":30,"time":"2023-07-30T00:51:52.912135756Z","hostname":"oxz_propolis-server_c08fc106-9de9-4679-959b-cb66fe71644d","pid":4713,"component":"crucible-6a9b2a6f-0501-4be4-9679-cceb4c893b09"}
{"msg":"[2] Received Imok","v":0,"name":"propolis-server","level":30,"time":"2023-07-30T00:51:54.47273673Z","hostname":"oxz_propolis-server_c08fc106-9de9-4679-959b-cb66fe71644d","pid":4713,"component":"crucible-6a9b2a6f-0501-4be4-9679-cceb4c893b09"}
```

**Comment by askfongjojo** [comment]
I recalled that this situation was observed at least twice in the past but unfortunately we didn't have a ticket for the issue. I'm checking if making disks as a backup can serve as a workaround (to avoid data loss at cold boot).

**Comment by jmpesp** [comment]
This sounds like a case where the read-only downstairs aren't being restarted after the zone bounces, aka https://github.com/oxidecomputer/crucible/pull/859.

Note this isn't a case of *data loss*, just that the downstairs 0 (in the log above) isn't started up.

**Comment by askfongjojo** [comment]
Yes, I noticed that I the downstairs SMF services had an empty log file after filing this bug. But prior to this fix, bouncing the SMF service for the 3 downstairs and rebooting the instance still didn't allow the guest to boot up. On hindsight, now I wonder if it would have worked if I stopped and started the instance instead of using the reboot operation.

Anyway, we've confirmed that the issue has been resolved by #859!
